pkg/controller/operators/olm/operator.go

```diff
@@ -855,15 +849,18 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		}
 
 		// Check for CRD ownership conflicts
-		csvSet := a.csvSet(out.GetNamespace(), v1alpha1.CSVPhaseAny)
-		if syncError = a.crdOwnerConflicts(out, csvSet); syncError != nil {
-			out.SetPhaseWithEvent(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonOwnerConflict, fmt.Sprintf("crd owner conflict: %s", syncError), now, a.recorder)
+		if syncError = a.crdOwnerConflicts(out, a.csvSet(out.GetNamespace(), v1alpha1.CSVPhaseAny)); syncError != nil {
+			if syncError == ErrCRDOwnerConflict {
```

Comment:
Is the idea here to protect against future additional errors being added to crdOwnerConflicts? The same thing is done with apiServiceOwnerConflicts too.

Reply:
Yup, even in the current implementation we can return transient errors that we don't want to fail on.